src/backend/src/controllers/change-requests.controllers.ts

```diff
@@ -1,509 +1,104 @@
-          detail: buildChangeDetail('Duration', String(workPackage.duration), String(updatedDuration))
-        }
-      ];
-      await prisma.project.update({
-        where: { projectId: workPackage.projectId },
-        data: {
-          budget: newBudget,
-          workPackages: {
-            update: {
-              where: { workPackageId: workPackage.workPackageId },
-              data: {
-                duration: updatedDuration,
-                wbsElement: {
-                  update: {
-                    changes: {
-                      create: changes[1]
-                    }
-                  }
-                }
-              }
-            }
-          },
-          wbsElement: {
-            update: {
-              changes: {
-                create: changes[0]
-              }
-            }
-          }
-        }
-      });
+import { Request, Response, NextFunction } from 'express';
+import ChangeRequestService from '../services/change-request.services';
+
+export default class CRController {
+  // Fetch the specific change request by its integer ID
+  static async getChangeRequestByID(req: Request, res: Response, next: NextFunction) {
+    try {
+      const crId: number = parseInt(req.params.crId);
+      const cr = await ChangeRequestService.getChangeRequestByID(crId);
+      return res.status(200).json(cr);
+    } catch (error: unknown) {
+      next(error);
     }
   }
-  // update change request
-  const updated = await prisma.change_Request.update({
-    where: { crId },
-    data: {
-      reviewer: { connect: { userId: reviewerId } },
-      reviewNotes,
-      accepted,
-      dateReviewed: new Date()
-    },
-    include: { activationChangeRequest: true, wbsElement: { include: { workPackage: true } } }
-  });
-
-  // verify wbs element exists
-  const wbsElement = await prisma.wBS_Element.findUnique({
-    where: {
-      wbsElementId: updated.wbsElementId
-    },
-    include: {
-      workPackage: true
+  static async getAllChangeRequests(req: Request, res: Response, next: NextFunction) {
+    try {
+      const changeRequests = await ChangeRequestService.getAllChangeRequests();
+      return res.status(200).json(changeRequests);
+    } catch (error: unknown) {
+      next(error);
     }
-  });
-
-  if (!wbsElement) {
-    return res.status(404).json({ message: `wbs element with id #${updated.wbsElementId} not found` });
   }
-
-  if (updated.accepted && foundCR.type === CR_Type.STAGE_GATE) {
-    const shouldChangeStatus = wbsElement.status !== WBS_Element_Status.COMPLETE;
-
-    const changesList = [];
-    if (shouldChangeStatus) {
-      changesList.push({
-        changeRequestId: crId,
-        implementerId: reviewerId,
-        detail: buildChangeDetail('status', wbsElement.status, WBS_Element_Status.COMPLETE)
-      });
+  // handle reviewing of change requests
+  static async reviewChangeRequest(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { body } = req;
```

Comment:
for all of these functions it should instead be:

`const { reviewerId, crId, reviewNotes, accepted, psId } = req.body;`